File: watir-rspec/pages/flight_page.rb
load File.dirname(__FILE__) + '/abstract_page.rb'

class FlightPage < AbstractPage

  def initialize(browser)
    super(browser, "Select Flight")
  end

  def select_depart_from(depart_from)
    browser.select_list(:name, "fromPort").select(depart_from)
  end

  def select_depart_month(depart_month)
    browser.select_list(:id, "departMonth").select(depart_month)
  end

  def select_depart_day(depart_day)
    browser.select_list(:name, "departDay").select(depart_day)
  end

  def select_arrive_at(arriving_in)
    browser.select_list(:name, "toPort").select(arriving_in)
  end

  def select_return_month(return_month)
    browser.select_list(:id, "returnMonth").select(return_month)
  end

  def select_return_day(return_day)
    browser.select_list(:name, "returnDay").select(return_day)
  end

  def select_service_class(service_class)
    browser.radio(:name, "servClass", service_class).click
  end

  def select_trip_type(trip_type)
    browser.radio(:name => "tripType", :value => trip_type).click
  end

  def click_continue
    browser.button(:value, "Continue").click
    PassengerPage.new(browser)
  end

end

File: selenium-webdriver-nodejs-mocha/pages/flight_page.js
const AbstractPage = require('./abstract_page');

var webdriver = require('selenium-webdriver'),
    By = webdriver.By,
    until = webdriver.until;

class FlightPage extends AbstractPage {

    constructor(driver) {
        super(driver);
    }

    selectTripType(trip_type) {
        this.driver.findElement(By.xpath("//input[@name='tripType' and @value='" + trip_type + "']")).click();
    }

    selectDepartFrom(city) {
        this.driver.findElement(By.name("fromPort")).sendKeys(city);
    }

    selectArriveAt(city) {
        this.driver.findElement(By.name("toPort")).sendKeys(city);
    }

	selectDepartDay(day) {
        this.driver.findElement(By.name("departDay")).sendKeys(day);
		
	}
	
	selectDepartMonth(month_year) {
        this.driver.findElement(By.name("departMonth")).sendKeys(month_year);		
	}
	
	selectReturnDay(day) {
        this.driver.findElement(By.name("returnDay")).sendKeys(day);		
	}
	
	selectReturnMonth(month_year) {
        this.driver.findElement(By.name("returnMonth")).sendKeys(month_year);
	}
      
	clickContinue() {
        this.driver.findElement(By.xpath("//input[@value='Continue']")).click();		
	}
};

module.exports = FlightPage;
